### ISSUE TITLE
Link prediction from time-series single cell data 

### ISSUE THREAD

**Issue by ThisisHubert** [benedekrozemberczki/pytorch_geometric_temporal#113]
Hi, this is my first experience using temporal graph for my research and I have a few questions. 

1. How many graph snapshots (dataset) are actually recommended to use the library? 

2. About the sample code [here](https://pytorch-geometric-temporal.readthedocs.io/en/latest/notes/introduction.html#epidemiological-forecasting), How can reconstruct the predicted graph using decoder from this point?  
```
from tqdm import tqdm

model = RecurrentGCN(node_features = 4)

optimizer = torch.optim.Adam(model.parameters(), lr=0.01)

model.train()

for epoch in tqdm(range(200)):
    cost = 0
    for time, snapshot in enumerate(train_dataset):
        y_hat = model(snapshot.x, snapshot.edge_index, snapshot.edge_attr)
        cost = cost + torch.mean((y_hat-snapshot.y)**2)
    cost = cost / (time+1)
    cost.backward()
    optimizer.step()
    optimizer.zero_grad()
```

3. I would like to create a custom dataset from a gene expression matrix where I will construct graphs from it with nodes = gene names & edges = correlation value (related to question no. 1, I only have 5 snapshots ).  
Is this categorized as DynamicGraphStaticSignal since only the edges change over time while the nodes remain the same?

Thank you in advance. 

Best, 

Hubert S

**Comment by benedekrozemberczki** [comment]
Hi there @ThisisHubert,

1. It depends on the problem. It could work with 2 snapshots.
2. You could output an embedding matrix instead of a vector and calculate H * H^T - you could apply an activation function to the output matrix if you want to have certain properties (e.g. row wise normalization).
3. Yes, it exactly sounds like that.

Can you star the repo and cite the research paper?

Bests,

Benedek

**Comment by ThisisHubert** [comment]
Hi @benedekrozemberczki , thanks for the response. 
Understood. I think that should work then.  
For number 2, could you point me to a reference (codebase, etc.) so I can have a better understanding 
The research is still a work in progress but will do as soon as I can.

**Comment by benedekrozemberczki** [comment]
It is a simple matrix product, the loss you need is similar to this one:

https://github.com/benedekrozemberczki/AttentionWalk

**Comment by ThisisHubert** [comment]
Understood. I get how the whole flow works now. Thank you! Will close this issue
